components/user-mgt/org.wso2.carbon.identity.user.pre.update.profile.action/src/main/java/org/wso2/carbon/identity/user/pre/update/profile/action/internal/execution/PreUpdateProfileRequestBuilder.java

```diff
@@ -0,0 +1,326 @@
+    public ActionExecutionRequest buildActionExecutionRequest(FlowContext flowContext,
+                                                              ActionExecutionRequestContext actionExecutionContext)
+            throws ActionExecutionRequestBuilderException {
+
+        UserActionContext userActionContext =
+                flowContext.getValue(UserActionContext.USER_ACTION_CONTEXT_REFERENCE_KEY, UserActionContext.class);
+        PreUpdateProfileAction preUpdateProfileAction = (PreUpdateProfileAction) actionExecutionContext.getAction();
+
+        return new ActionExecutionRequest.Builder()
+                .actionType(getSupportedActionType())
+                .event(getEvent(userActionContext, preUpdateProfileAction))
+                .build();
+    }
+
+    private Event getEvent(UserActionContext userActionContext, PreUpdateProfileAction preUpdateProfileAction)
+            throws ActionExecutionRequestBuilderException {
+
+        PreUpdateProfileEvent.Builder eventBuilder = new PreUpdateProfileEvent.Builder();
+        eventBuilder.initiatorType(getInitiatorType());
+        eventBuilder.action(PreUpdateProfileEvent.Action.UPDATE);
+        eventBuilder.request(getPreUpdateProfileRequest(userActionContext));
+        eventBuilder.tenant(getTenant());
+
+        UniqueIDUserStoreManager userStoreManager = getUserStoreManager();
+        eventBuilder.user(getUser(userActionContext, preUpdateProfileAction, userStoreManager));
+        eventBuilder.userStore(getUserStore(userActionContext, userStoreManager));
+
+        return eventBuilder.build();
+    }
+
+    private PreUpdateProfileEvent.FlowInitiatorType getInitiatorType() throws ActionExecutionRequestBuilderException {
+
+        Flow flow = Optional.ofNullable(IdentityContext.getThreadLocalIdentityContext().getFlow())
+                .orElseThrow(() -> new ActionExecutionRequestBuilderException("Unknown flow."));
+
+        switch (flow.getInitiatingPersona()) {
+            case ADMIN:
+                return PreUpdateProfileEvent.FlowInitiatorType.ADMIN;
+            case APPLICATION:
+                return PreUpdateProfileEvent.FlowInitiatorType.APPLICATION;
+            case USER:
+                return PreUpdateProfileEvent.FlowInitiatorType.USER;
+            default:
+                throw new ActionExecutionRequestBuilderException(
+                        "Unknown initiator type for the flow: " + flow.getName());
+        }
+    }
+
+    private PreUpdateProfileRequest getPreUpdateProfileRequest(UserActionContext userActionContext)
+            throws ActionExecutionRequestBuilderException {
+
+        UserActionRequestDTO userActionRequestDTO = userActionContext.getUserActionRequestDTO();
+        PreUpdateProfileRequest.Builder preUpdateProfileRequestBuilder = new PreUpdateProfileRequest.Builder();
+
+        if (userActionRequestDTO.getClaims() == null || userActionRequestDTO.getClaims().isEmpty()) {
+            return preUpdateProfileRequestBuilder.build();
```

Comment:
By doing this will result an empty request object in the request payload right ?
Expected return should be `null` here.

Reply:
We are ignoring both null and empty values. So this does not go as an empty object.
But I'm also thinking we might need to well define these. There are cases where we want to communicate the empty object to the extension. In such cases, having both modes rejected can be problematic.

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6620/commits/709e04565c6a28dab6c76148d911afc62cf22db2